### PR TITLE
feat(drawer): use the breakpoint mods prop instead of width props

### LINKED ISSUE #4238: Update DrawerPanelContent to use breakpoint mods
DrawerPanelContent.tsx should change its API from:
```tsx
  /** Default width for drawer panel */
  width?: 25 | 33 | 50 | 66 | 75 | 100;
  /** Drawer panel width on large viewports */
  widthOnLg?: 25 | 33 | 50 | 66 | 75 | 100;
  /** Drawer panel width on xl viewports */
  widthOnXl?: 25 | 33 | 50 | 66 | 75 | 100;
  /** Drawer panel width on 2xl viewports */
  widthOn2Xl?: 25 | 33 | 50 | 66 | 75 | 100;
```
to
```tsx
  breakpointMods?: DrawerBreakpointMod[]
```
and use our `formatBreakpointMods` helper function.

### PR DESCRIPTION
**Pull request by boaz0**
**What**: Closes #4238 

//cc @redallen

### CODE REVIEW

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx [hunk 1]
```diff
@@ -11,42 +43,28 @@ export interface DrawerPanelContentProps extends React.HTMLProps<HTMLDivElement>
...
-    )}
-  </DrawerContext.Consumer>
-);
+}: DrawerPanelContentProps) => {
+  const t = formatBreakpointMods(breakpointMods, styles);
+  return (
+    <DrawerContext.Consumer>
+      {({ isExpanded, isStatic }) => (
+        <div
+          className={css(styles.drawerPanel, hasNoBorder && styles.modifiers.noBorder, t, className)}
```
Comment (redallen) [inline packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:61]: Sorry to be a style nit, but can you just replace `t` with `formatBreakpointMods(breakpointMods, styles);`?

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx [hunk 2]
```diff
@@ -2,6 +2,38 @@ import * as React from 'react';
...
+} as const;
+
+export const DrawerBreakpoints = {
+  none: '',
+  lg: 'lg',
+  xl: 'xl',
+  '2xl': '2xl'
+} as const;
+
+type ValueOf<T> = T[keyof T];
```
Comment (redallen) [inline packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:23]: Unused type

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx [hunk 3]
```diff
@@ -2,6 +2,38 @@ import * as React from 'react';
...
+  75: 'width_75',
+  100: 'width_100'
+} as const;
+
+export const DrawerBreakpoints = {
+  none: '',
+  lg: 'lg',
+  xl: 'xl',
+  '2xl': '2xl'
+} as const;
```
Comment (redallen) [inline packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:21]: Don't think you need `as const`? Maybe just make an enum?

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx [hunk 4]
```diff
@@ -2,6 +2,38 @@ import * as React from 'react';
...
+import { formatBreakpointMods } from '../../helpers/util';
+
+export const DrawerModifiers = {
+  25: 'width_25',
+  33: 'width_33',
+  50: 'width_50',
+  66: 'width_66',
+  75: 'width_75',
+  100: 'width_100'
+} as const;
```
Comment (redallen) [inline packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:14]: Don't think you need `as const`? Maybe just make an enum?

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx [hunk 5]
```diff
@@ -2,6 +2,38 @@ import * as React from 'react';
...
+  modifier:
+    | 'width_25'
+    | 'width_33'
+    | 'width_50'
+    | 'width_66'
+    | 'width_75'
+    | 'width_100'
+    | typeof DrawerModifiers[keyof typeof DrawerModifiers];
+  /** The breakpoint at which to apply the modifier */
+  breakpoint: '' | 'lg' | 'xl' | '2xl' | typeof DrawerBreakpoints[keyof typeof DrawerBreakpoints];
```
Comment (redallen) [inline packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:35]: If you make `DrawerBreakpoints` an enum you can just do `| DrawerBreakpoints`

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx (hunk 4 above)
Comment (boaz0) [reply packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:14]: Enum key cannot be a numerical value. :sob:

File: packages/react-core/src/helpers/util.ts [hunk 6]
```diff
@@ -244,6 +245,7 @@ export function pluralize(i: number, singular: string, plural?: string) {
  */
```
Comment (nicolethoen) [inline packages/react-core/src/helpers/util.ts:245]: Need to update the function comment

File: packages/react-core/src/helpers/util.ts [hunk 7]
```diff
@@ -255,6 +257,7 @@ export const formatBreakpointMods = (
   breakpointMods
     .map(mod => `${mod.modifier}${mod.breakpoint ? `-on-${mod.breakpoint}` : ''}`)
     .map(toCamel)
+    .map(mod => mod.replace(/-2xl$/gi, '_2xl'))
```
Comment (redallen) [inline packages/react-core/src/helpers/util.ts:260]: Thanks for fixing this! I've been forgetting to go back and do it.

**Comment by boaz0** [comment]
Thank you so much! @nicolethoen @redallen :smile_cat:

**Pull request merged**